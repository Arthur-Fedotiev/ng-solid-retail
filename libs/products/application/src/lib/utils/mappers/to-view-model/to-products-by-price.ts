import { ProductByPrice, ProductViewModel } from '../../../models';

export const toProductsByPrice = (
  acc: ReadonlyArray<ProductByPrice>,
  { prices, id, name, sku, url }: ProductViewModel
): ReadonlyArray<ProductByPrice> => {
  const productsForPrice = prices.map(({ value: price, retailer }) => ({
    price: { price, retailer: { name: retailer } },
    id,
    name,
    sku,
    url,
  }));

  return [...acc, ...productsForPrice];
};
