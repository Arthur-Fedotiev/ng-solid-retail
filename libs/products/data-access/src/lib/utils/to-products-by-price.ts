import { ProductByPrice } from '../models/product-by-price.type';
import { ProductViewModel } from '../models/ProductViewModel';

export const toProductsByPrice = (
  acc: ReadonlyArray<ProductByPrice>,
  { prices, id, name, sku, url }: ProductViewModel
): ReadonlyArray<ProductByPrice> => {
  const productsForPrice = prices.map(({ price, retailer }) => ({
    price: { price, retailer: { name: retailer.name } },
    id,
    name,
    sku,
    url,
  }));

  return [...acc, ...productsForPrice];
};
