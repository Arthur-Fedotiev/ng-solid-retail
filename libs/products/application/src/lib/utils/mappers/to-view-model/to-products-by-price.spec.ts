import { makeProductViewModelsStub } from '../../../testing/make-product-view-models-stub';
import { toProductsByPrice } from './to-products-by-price';

describe('#toProductsByPrice', () => {
  it('should return a ProductByPrice', () => {
    const productsStub = makeProductViewModelsStub(2);
    const result = productsStub.reduce(toProductsByPrice, []);

    expect(result).toMatchSnapshot();
  });
});
