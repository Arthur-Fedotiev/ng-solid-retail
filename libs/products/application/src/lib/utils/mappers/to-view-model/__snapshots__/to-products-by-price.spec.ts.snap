// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#toProductsByPrice should return a ProductByPrice 1`] = `
[
  {
    "id": "0",
    "name": "Product 0",
    "price": {
      "price": 0,
      "retailer": {
        "name": "Retailer 1",
      },
    },
    "sku": "12345",
    "url": "Product 0 url",
  },
  {
    "id": "0",
    "name": "Product 0",
    "price": {
      "price": 0,
      "retailer": {
        "name": "Retailer 2",
      },
    },
    "sku": "12345",
    "url": "Product 0 url",
  },
  {
    "id": "0",
    "name": "Product 0",
    "price": {
      "price": 0,
      "retailer": {
        "name": "Retailer 3",
      },
    },
    "sku": "12345",
    "url": "Product 0 url",
  },
  {
    "id": "1",
    "name": "Product 1",
    "price": {
      "price": 1,
      "retailer": {
        "name": "Retailer 1",
      },
    },
    "sku": "12345",
    "url": "Product 1 url",
  },
  {
    "id": "1",
    "name": "Product 1",
    "price": {
      "price": 2,
      "retailer": {
        "name": "Retailer 2",
      },
    },
    "sku": "12345",
    "url": "Product 1 url",
  },
  {
    "id": "1",
    "name": "Product 1",
    "price": {
      "price": 3,
      "retailer": {
        "name": "Retailer 3",
      },
    },
    "sku": "12345",
    "url": "Product 1 url",
  },
]
`;
